### PR TITLE
feat(transfer): add `logging_config` to `google_storage_transfer_job`

### PR DESCRIPTION
**Pull request by AntoineDao**
This PR adds the `logging_config` field to the `google_storage_transfer_job` resources. I'm not sure I could find an issue for this here but I did raise one in https://github.com/hashicorp/terraform-provider-google/issues/12802 and an associate PR (which should probably be closed) here: https://github.com/hashicorp/terraform-provider-google/pull/12803

This fixes https://github.com/hashicorp/terraform-provider-google/issues/12802

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
storagetransfer: added `logging_config` to `google_storage_transfer_job`
```

### CODE REVIEW

**Comment by AntoineDao** [comment]
Hope you don't mind reviewing the code while I sort out CLA on my end 😄

**Comment by AntoineDao** [comment]
Hi @shuyama1, I could do with a bit of help understanding what is broken with my changes as I don't have access to the CloudBuild logs... 

I tried following the instructions to run tests locally by generating the terraform provider with the following command:
> ./scripts/make-in-container.sh terraform VERSION=ga OUTPUT_PATH="$GOPATH/src/github.com/hashicorp/terraform-provider-google" however so many of the tests have been marked as failing that I'm not sure which one to start with 😓

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 1]
```diff
@@ -172,6 +172,42 @@ func resourceStorageTransferJob() *schema.Resource {
...
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{"SUCCEEDED", "FAILED"}, false),
+							},
+							Description: `States in which logActions are logged. Not supported for transfers with PosifxFilesystem data sources; use enable_on_prem_gcs_transfer_logs instead.`,
+						},
+						"enable_on_prem_gcs_transfer_logs": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							ConflictsWith: []string{"logging_config.log_actions,logging_config.log_action_states"},
```
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:204]:
```suggestion
							ConflictsWith: []string{"logging_config.0.log_actions", "logging_config.0.log_action_states"},
```

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 2]
```diff
@@ -172,6 +172,42 @@ func resourceStorageTransferJob() *schema.Resource {
...
+			"logging_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"log_actions": {
+							Type:          schema.TypeList,
+							Optional:      true,
+							ConflictsWith: []string{"logging_config.enable_on_prem_gcs_transfer_logs"},
```
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:184]:
```suggestion
							ConflictsWith: []string{"logging_config.0.enable_on_prem_gcs_transfer_logs"},
```

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 3]
```diff
@@ -172,6 +172,42 @@ func resourceStorageTransferJob() *schema.Resource {
...
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{"FIND", "DELETE", "COPY"}, false),
+							},
+							Description: `Specifies the actions to be logged. Not supported for transfers with PosifxFilesystem data sources; use enable_on_prem_gcs_transfer_logs instead.`,
+						},
+						"log_action_states": {
+							Type:          schema.TypeList,
+							Optional:      true,
+							ConflictsWith: []string{"logging_config.enable_on_prem_gcs_transfer_logs"},
```
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:194]:
```suggestion
							ConflictsWith: []string{"logging_config.0.enable_on_prem_gcs_transfer_logs"},
```

**Review by shuyama1 — CHANGES_REQUESTED** [review]
I think the format of `ConflictsWith` is a bit off. Modifying it should get the test run. I'll start reviewing the rest of the code after I see the test results.

**Comment by AntoineDao** [comment]
Good catch on the `ConflictsWith` bug! Looks like the tests are all passing, except for the VRC (not sure what it is 😅 ) and the CLA one which I'm still chasing with my employer. Overall are there any changes you think we should include in this PR?

**Comment by shuyama1** [comment]
VCR test is the one we run all the acceptance tests against your PR change, and it looks like the provider crashed while running the tests due to the error:
```
panic: interface conversion: interface {} is []interface {}, not []string

goroutine 1455835 [running]:
github.com/hashicorp/terraform-provider-google-beta/google-beta.expandTransferJobLoggingConfig({0xc0060e0190?, 0x3520c45?, 0x3d85040?})
	/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_storage_transfer_job.go:1215 +0x234
github.com/hashicorp/terraform-provider-google-beta/google-beta.resourceStorageTransferJobCreate(0x0?, {0x34e3c20?, 0xc003034400})
	/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_storage_transfer_job.go:577 +0x2ed
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x3d7ecd0?, {0x3d7ecd0?, 0xc0145f9680?}, 0xd?, {0x34e3c20?, 0xc003034400?})
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc01d561c00, {0x3d7ecd0, 0xc0145f9680}, 0xc00acaca90, 0xc0034bd980, {0x34e3c20, 0xc003034400})
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0121ca000, {0x3d7ecd0?, 0xc0145f9620?}, 0xc021bdc910)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc008d841e0, {0x3d7ecd0?, 0xc0145f9080?}, 0xc002aa9f80)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:813 +0x4fc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x346dec0?, 0xc008d841e0}, {0x3d7ecd0, 0xc0145f9080}, 0xc002aa9f10, 0x0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc005a9c1e0, {0x3d85100, 0xc01c814b60}, 0xc0179e4d80, 0xc01f493a10, 0x538d580, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1301 +0xb0b
google.golang.org/grpc.(*Server).handleStream(0xc005a9c1e0, {0x3d85100, 0xc01c814b60}, 0xc0179e4d80, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1642 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:938 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:936 +0x28a
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	1700.465s
FAIL
```

**Comment by AntoineDao** [comment]
Great thanks for expediting the logs my way. I'll aim to fix it over the next few days and ping you once it's ready for review 😄

**Comment by AntoineDao** [comment]
Hi @shuyama1, I think this if *finally* ready for review 😄

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 4]
```diff
@@ -1153,3 +1204,45 @@ func flattenTransferJobNotificationConfig(notificationConfig *storagetransfer.No
...
+	LogActionStates := make([]string, len(LogActionStatesInterface))
+	for i, v := range LogActionStatesInterface {
+		LogActionStates[i] = fmt.Sprint(v)
+	}
+
+	var apiData = &storagetransfer.LoggingConfig{
+		LogActions:                  logActions,
+		LogActionStates:             LogActionStates,
+		EnableOnpremGcsTransferLogs: loggingConfig["enable_on_prem_gcs_transfer_logs"].(bool),
+	}
```
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:1230]: This looks good, but for simplicity, we can just use `convertStringArr`, such as 
```suggestion
	var apiData = &storagetransfer.LoggingConfig{
		LogActions:                  convertStringArr(loggingConfig["log_actions"].([]interface{})),
		LogActionStates:             convertStringArr(loggingConfig["log_action_states"].([]interface{})),
		EnableOnpremGcsTransferLogs: loggingConfig["enable_on_prem_gcs_transfer_logs"].(bool),
	}
```

**Comment by AntoineDao** [comment]
@shuyama1 I think that should be all of your comments addressed 😄

**Review by shuyama1 — CHANGES_REQUESTED** [review]

**Comment by michellepolte-optimizely** [comment]
Hey, are there any updates to this feature? The logging configuration whould be important for us.

**Pull request closed without merging**